### PR TITLE
Add Forge foundry

### PR DESCRIPTION
**Pull request by ferranbt**
## 📝 Summary

This PR adds some basic support to debug Suave apps in Solidity with Foundry. It introduces a generated Suave contract that implements the precompiles by making `vm.ffi` calls to an external suave command to execute the precompiles.

## 📚 References

---

* [x] I have seen and agree to CONTRIBUTING.md

### CODE REVIEW

File: internal/ethapi/api.go [hunk 1]
```diff
@@ -1944,7 +1944,7 @@ func (s *TransactionAPI) SendRawTransaction(ctx context.Context, input hexutil.B
 }
 
 // TODO: should be its own api
-func runMEVM(ctx context.Context, b Backend, state *state.StateDB, header *types.Header, tx *types.Transaction, msg *core.Message) (*types.Transaction, *core.ExecutionResult, func() error, error) {
+func runMEVM(ctx context.Context, b Backend, state *state.StateDB, header *types.Header, tx *types.Transaction, msg *core.Message, isCall bool) (*types.Transaction, *core.ExecutionResult, func() error, error) {
```
Comment (Ruteri) [inline internal/ethapi/api.go:1947]: Nit: would it be clearer to pass in the `vm.Config`?

Comment (ferranbt) [reply internal/ethapi/api.go:1947]: For now, there is no other field in `vm.Config` that we want to configure. I would do it if that is the case.

File: core/vm/contracts_suave.go [hunk 2]
```diff
@@ -123,6 +124,8 @@ func (c *confStoreStore) runImpl(suaveContext *SuaveContext, bidId suave.BidId,
 		return suave.ErrBidNotFound
 	}
 
+	log.Info("confStoreStore", "bidId", bidId, "key", key)
```
Comment (Ruteri) [inline core/vm/contracts_suave.go:127]: probably not needed

**Review by Ruteri — APPROVED** [review]

File: core/vm/contracts_suave.go (hunk 2 above)
Comment (ferranbt) [reply core/vm/contracts_suave.go:127]: We do have the same log for retrieve, do you want to remove that one too?

Comment (Ruteri) [reply core/vm/contracts_suave.go:127]: I think so

**Review by Ruteri — APPROVED** [review]

**Pull request merged**